platform/openide.util.ui/src/org/openide/util/Utilities.java

```diff
@@ -1357,17 +1357,50 @@ private static Frame findMainWindow()
      */
     // PR4739
     public static Component findDialogParent() {
-        Component parent = KeyboardFocusManager.getCurrentKeyboardFocusManager().getFocusOwner();
+        return findDialogParent(null);
+    }
+
+    /**
+     * Finds an appropriate component to use for a dialog's parent. Similar to {@link #findDialogParent()}
+     * with the ability to specify a suggested parent component.
+     *
+     * @param suggestedParent the component to return if suitable
+     * @return A suitable parent component for swing dialog displayers.
+     *
+     * @see #findMainWindow()
+     * @since 9.30
+     */
+    public static Component findDialogParent(Component suggestedParent) {
+        Component parent = suggestedParent;
         if (parent == null) {
-            parent = KeyboardFocusManager.getCurrentKeyboardFocusManager().getActiveWindow();
+            parent = KeyboardFocusManager.getCurrentKeyboardFocusManager().getFocusOwner();
+        }
+        Window active = KeyboardFocusManager.getCurrentKeyboardFocusManager().getActiveWindow();
+        if (parent == null) {
+            parent = active;
+        } else if (active instanceof Dialog && ((Dialog) active).isModal()) {
+            Window suggested = parent instanceof Window ? (Window) parent : SwingUtilities.windowForComponent(parent);
+            if (suggested != active) {
+                return active;
+            }
         }
         if (parent == null) {
             // PR#5280
-            parent = findMainWindow();
+            return findMainWindow();
         }
         return parent;
     }
 
+    /**
+     * Gets whether a modal dialog is open.
+     * @return true if a modal dialog is open, false otherwise
+     * @since 9.30
+     */
+    public static boolean isModalDialogOpen() {
```

Comment:
I guess `isModaldialogOpen` is here, even though it's only used from one spot, to keep this stuff together/centralized. Remember to document any API changes; this one in `platform/openide.util.ui/apichanges.xml`.

Reply:
Yes, this seemed like a general purpose method so adding it to `Utilities` seemed reasonable.

> Remember to document any API changes

Will do; thanks for the reminder.

Reply:
> Yes, this seemed like a general purpose method so adding it to `Utilities` seemed reasonable.

I'm not an *official reviewer* and decisions about what belongs in the API are above my pay grade, but I'm always willing (too willing some might say) to offer an opinion. I believe that most stuff in API is about NetBeans; I'm not sure that convenience methods that aren't NB related belong in the public API, for example `isModalDialogOpen()`.

Reply:
It doesn't seem like a stretch to have it here, but I can make it private to `NodeOperationImpl` if that's more palatable.

Reply:
IMO it seems a useful method.  Not sure if it needs to iterate up the window ownership tree though?